pkg/skaffold/deploy/deploy.go

```diff
@@ -21,15 +21,26 @@ import (
 	"fmt"
 	"io"
 
+	"k8s.io/apimachinery/pkg/runtime"
+
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
 )
 
+// Artifact contains all information about a completed deployment
+type Artifact struct {
+	obj       *runtime.Object
+	namespace string
```

Comment:
why we need namespace here?

Reply:
the namespace used to do a deployment doesn't always match the namespace in the deployer, and we need access to it when updating the API objects with the labels